firmware/ValdisFirmwareStuff/Core/Src/main.c

```diff
@@ -0,0 +1,771 @@
+			pwmValue = PID(&M1, pos);
+			HAL_GPIO_WritePin(M1_DIR_A_GPIO_Port, M1_DIR_A_Pin, (pwmValue < 0) ? GPIO_PIN_SET : GPIO_PIN_RESET);
+			TIM2->CCR1 =(uint16_t)clmp((pwmValue < 0) ? -pwmValue : pwmValue, 0 , 65535);
+
+			pos = (int16_t)TIM3->CNT;
+			pwmValue = PID(&M2, pos);
+			HAL_GPIO_WritePin(M2_DIR_A_GPIO_Port, M2_DIR_A_Pin, (pwmValue < 0) ? GPIO_PIN_SET : GPIO_PIN_RESET);
+			TIM2->CCR2 =(uint16_t)clmp((pwmValue < 0) ? -pwmValue : pwmValue, 0 , 65535);
+
+			pos = (int16_t)TIM4->CNT;
+			pwmValue = PID(&M3, pos);
+			HAL_GPIO_WritePin(M3_DIR_A_GPIO_Port, M3_DIR_A_Pin, (pwmValue < 0) ? GPIO_PIN_SET : GPIO_PIN_RESET);
+			TIM2->CCR3 =(uint16_t)clmp((pwmValue < 0) ? -pwmValue : pwmValue, 0 , 65535);
+
+			toggle_nSleep();
+
+
+		  // Motor control calculations can be called from here
+		}
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USB_Device_Init();
+  MX_TIM1_Init();
+  MX_TIM2_Init();
+  MX_TIM3_Init();
+  MX_TIM4_Init();
+  MX_TIM15_Init();
+  MX_TIM6_Init();
+  /* USER CODE BEGIN 2 */
+
+
+
+  //start motor PWM
+
+
+
+    HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_1);
+    HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_2);
+    HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_3);
+
+    HAL_TIM_PWM_Start(&htim15, TIM_CHANNEL_2);
+
+    TIM2->CCR1 = 0;
+    TIM2->CCR2 = 0;
+    TIM2->CCR3 = 0;
+    //HAL_Delay(100);
+    //TIM15->CCR2 = 3000;
+    TIM15->CCR2 = 0;
+
+
+    toggle_nSleep();
+
+    //start encoders
+        HAL_TIM_Encoder_Start(&htim1, TIM_CHANNEL_1 | TIM_CHANNEL_2);
+        HAL_TIM_Encoder_Start(&htim3, TIM_CHANNEL_1 | TIM_CHANNEL_2);
+        HAL_TIM_Encoder_Start(&htim4, TIM_CHANNEL_1 | TIM_CHANNEL_2);
+
+
+        Feedback feedback = { // (1)
+        	        .speed1 = 0,
+        	        .speed2 = 0,
+        	        .speed3 = 0,
+        	        .delimiter = 0xAAAA
+        	    };
+
+
+        //Thrower_Send(0);
+    HAL_TIM_Base_Start_IT(&htim6);
+
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+
+    /* USER CODE END WHILE */
+
+    /* USER CODE BEGIN 3 */
+
+  if (isCommandReceived) {
+		isCommandReceived = 0;
+		toggle_nSleep();
+		HAL_GPIO_TogglePin(DEBUG_LED_GPIO_Port, DEBUG_LED_Pin);
+
+		M1.speed = command.speed1;
+		M2.speed = command.speed2;
+		M3.speed = command.speed3;
+
+		TIM15->CCR2 = command.throwerSpeed;
+
+
+		feedback.speed1 = M1.speed;
+		feedback.speed2 = M2.speed;
+		feedback.speed3 = M3.speed;
```

Comment:
It would be more useful to send `positionChange` with feedback, instead of the same speed values from the command.